.github/renovate.json

```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:base"
+  ],
+  "prConcurrentLimit": 5,
+  "baseBranches": ["master","release-1.8","release-1.9","release-1.10"],
+  "ignorePaths": ["design/**"],
+  "postUpdateOptions": ["gomodTidy"],
+  "recreateClosed": true,
+  "regexManagers": [
+    {
+      "fileMatch": ["^.github/workflows/ci.yml$"],
+       "matchStrings": [
+        "#\\s*renovate:\\s*datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\s.*?_VERSION='(?<currentValue>.*)'$"
+      ],
+      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
+    }
+  ],
+  "packageRules": [
+    {
+      "matchDatasources": [
+        "go"
+      ],
+      "groupName": "kubernetes non-major packages",
+      "groupSlug": "kubernetes-go",
+      "matchDepNames": [
+        "k8s.io/client-go"
+      ],
+      "allowedVersions": "<1.0"
+    }, {
+      "matchDatasources": [
+        "go"
+      ],
+      "groupName": "kubernetes non-major packages",
+      "groupSlug": "kubernetes-go",
+      "matchUpdateTypes": [
+        "minor",
+        "patch"
+      ],
+      "matchPackagePrefixes": [
```

Comment:
How does separating out the k8s deps from the other deps (both of which we want minor and patch bumps on) impact renovate? Feel free to point me to docs if more efficient!

Reply:
https://docs.renovatebot.com/configuration-options/#configuration-options.

They don't seem to state explicitly but it looks like this config file is actually https://json5.org (which I had never heard of before) - a superset of JSON that supports `//` style comments. I definitely encourage dropping as much context/rationale into config files like this as possible so that future readers can understand intent.

Reply:
Added comments, thanks for pointing that out!